collect_app/src/test/java/org/odk/collect/android/database/DatabaseInstancesRepositoryTest.kt

```diff
@@ -34,4 +38,32 @@ class DatabaseInstancesRepositoryTest : InstancesRepositoryTest() {
     override fun getInstancesDir(): String {
         return instancesDir.absolutePath
     }
+
+    @Test
+    fun save_failsWhenEditOfPointsAtItsOwnDbId() {
```

Comment:
I'd be tempted to have these tests in `InstancesRepositoryTest` with either a less granular exception assertion, or a wrapper exception (`IntegrityException` came to me off the top of my head for example). That way we keep the fake as close to the "real" implementation as possible avoiding any behaviour mismatches in unit tests and real use.

Reply:
Done.